src/REL/d/a/obj/d_a_obj_pool_cock.cpp

```diff
@@ -0,0 +1,107 @@
+#include <d/a/obj/d_a_obj_pool_cock.h>
+#include <s/s_Math.h>
+#include <toBeSorted/sceneflag_manager.h>
+
+SPECIAL_ACTOR_PROFILE(OBJ_POOL_COCK, dAcOPoolCock_c, fProfile::OBJ_POOL_COCK, 0x024D, 0, 7);
+
+STATE_DEFINE(dAcOPoolCock_c, Wait);
+
+bool dAcOPoolCock_c::createHeap() {
+    mBrres = getOarcResFile("WaterD101");
+    RoomManager::bindStageResToFile(&mBrres);
+    nw4r::g3d::ResMdl mdl = mBrres.GetResMdl("PoolCockD101");
+    for (int i = 0; i < 2; i++) {
+        TRY_CREATE(mModels[i].create(mdl, &heap_allocator, 0x120));
+    }
+    return true;
+}
+
+int dAcOPoolCock_c::actorCreate() {
+    CREATE_ALLOCATOR(dAcOPoolCock_c);
+
+    forwardAccel = -0.0f;
+    forwardMaxSpeed = -40.0f;
+    mOpenSceneflag = params & 0xFF;
+    mOpenDirection = mVec3_c::Ez;
+    mOpenDirection.rotY(rotation.y);
+
+    if (mOpenSceneflag < 0xFF && SceneflagManager::sInstance->checkBoolFlag(roomid, mOpenSceneflag)) {
+        mOpenProgress = 400.0f;
+    }
+
+    mStateMgr.changeState(StateID_Wait);
+
+    setBoundingBox(mVec3_c(-300.0f, -100.0f, -300.0f), mVec3_c(300.0f, 100.0f, 300.0f));
+    return SUCCEEDED;
+}
+
+int dAcOPoolCock_c::actorPostCreate() {
+    return SUCCEEDED;
+}
+
+int dAcOPoolCock_c::doDelete() {
+    return SUCCEEDED;
+}
+
+int dAcOPoolCock_c::actorExecute() {
+    mStateMgr.executeState();
+    calcVelocity();
+    position += velocity;
+    position += posIncrements;
+    updateMatrix();
+    mMtx_c mdl1Transform;
+    mMtx_c mdl2Transform;
+    mdl2Transform = mdl1Transform = worldMatrix;
```

Comment:
While this seems to be the key to matching this function and the copy assignment operator breaks that. Note the `mdl2Transform.YrotM` call below that gives us concrete evidence that mMtx_c is being used here.